### DISCUSSION TITLE
🎧 Kaizen 10 - The best pipeline ever™️ - Jun. 2, 2023

### DISCUSSION THREAD

**Discussion by gerhard** [thechangelog/changelog.com#452] [Kaizen]
[![shields.io](https://img.shields.io/badge/Last%20updated%20on%20July%201%2C%202023-PUBLISHED-blue?style=for-the-badge)](https://changelog.com/friends/2)

[🎧 This is now available as changelog.com/friends/2](https://changelog.com/friends/2)

<a href="https://changelog.com/friends/2"><img width="1440" alt="image" src="https://github.com/thechangelog/changelog.com/assets/3342/bb84b75a-93e1-441e-82a5-a555aa9f818c"></a>

See all improvements in this 🐙 GitHub repository: [Kaizen 9 ... 10](https://github.com/thechangelog/changelog.com/compare/kaizen.9...kaizen.10)

---

This contains all things related to the next Kaizen episode.

Share your thoughts via comments below:
- Is there something in changelog.com that you, our dear listener, would like us to tackle?
- Is there something that **you** would like to help out with?
- Anything that we - @adamstac @jerodsanto - want to improve?

This is a rough timeline that we have in mind (red line is today):

```mermaid
gantt
    title 🎧 Kaizen 10 timeline
    dateFormat  YYYY-MM-DD
    axisFormat %b %e, %Y

    section DISCUSS
    ✍️ Add your comments below :2023-03-07, 77d

    section CONTRIBUTE
    💻 ALL CONTRIBUTIONS WELCOME 👕 :2023-03-07, 77d

    section RECORD
    🎙 Record :milestone, 2023-05-23
    ✂️ Edit :2023-05-23, 10d
    🚀 SHIP IT! :milestone, 2023-06-02

    section NEXT
    Kick-off Kaizen 11 :2023-06-02, 14d
```

---

### 🎙 Proposed topics

- [x] #454
  - Removed the last Makefile
- [x] #455 - Erlang, Elixir & Node.js + `flyctl` & Dagger Go SDK
  - 20 mins - including local CI testing
  - 🙏 Thank you @jamietanna!
  - BTW: [`.tools-versions`](https://github.com/thechangelog/changelog.com/blob/34d9a65a3730975fe78633c22191d4003d08cb35/.tool-versions) via [asdf-vm](https://asdf-vm.com/) - Manage multiple runtime versions with a single CLI tool
- [x] #464
- [x] Let me show you something cool
  - [Changelog weekly issue #451 • 2023-04-09](https://email.changelog.com/t/t-e-futakk-yhjdttkyk-nt/) - the last one
    - martinheinz.dev - [Make Your CLI Demos a Breeze with Zero Stress and Zero Mistakes](https://martinheinz.dev/blog/94)
    - [saschagrunert/demo](https://github.com/saschagrunert/demo) - A framework for pre-recorded CLI demos
```console
brew upgrade && brew install dagger/homebrew-tap/dagger
dagger version
# must be >= 0.5.3
export _EXPERIMENTAL_DAGGER_TUI=inline
dagger run mage ci
```
- [x] Typesense
- [x] Fly.io [Reliability: It’s Not Great](https://community.fly.io/t/reliability-its-not-great/11253) - Reliability, it **is** great for changelog.com. @gerhard to pull stats from Honeycomb.
<img width="2880" alt="image" src="https://user-images.githubusercontent.com/3342/232123750-315ff088-c873-4a07-99a6-81b65b6a4903.png">

- [x] #456

---

### What changed so far? [Kaizen.9 ... Kaizen.10](https://github.com/thechangelog/changelog.com/compare/kaizen.9...kaizen.10)
![image](https://github.com/thechangelog/changelog.com/assets/3342/2518ac0d-47e8-45ea-8522-c61993bd53fe)
- https://github.com/thechangelog/changelog.com/pull/454
- https://github.com/thechangelog/changelog.com/pull/455
- https://github.com/thechangelog/changelog.com/pull/456
- https://github.com/thechangelog/changelog.com/pull/462 🙏 Thank you @ken-kost 
- https://github.com/thechangelog/changelog.com/pull/464

### What else is coming up?
- https://github.com/thechangelog/changelog.com/pull/451
- 1Password Connect Server on Fly.io
- Honeycomb SLOs replace Pingdom monitoring
  - Uptime Kuma experiment
  - https://github.com/thechangelog/changelog.com/issues/457#issuecomment-1503027429

---

This is a follow-up to:
- https://github.com/thechangelog/changelog.com/discussions/440
- https://github.com/thechangelog/changelog.com/discussions/433

**Comment by jerodsanto** [comment]
My main goal for this Kaizen is spinning off Changelog News into its own podcast + newsletter that replaces Changelog Weekly. This requires some technical changes/improvements to the app as well as a migration that will also require some single-use code. **I feel confident that this will get done prior to Kaizen 10**.

My secondary goal is to land on and execute a new plan for our feeds (I discussed this some on Kaizen 9 and the discussion with listeners of that is ongoing) that allows for: 

1. Consistently fast _enough_ responses
2. Clustering of app servers
3. Instant updates on episode publish/edit
4. Scaling from 10ish feeds to hundreds or low thousands

**I feel not as confident that this will get done prior to Kiazen 10**. In fact, I might just wait until we have Honeycomb visibility into app responses before I make serious moves on this effort...

In addition to all that, we have the version 0 of our Typesense integration finished. We should probably have Jason Bosco on (at least for a subset) to discuss his work on Kaizen 10.

**Reply by gerhard** [reply]
I am starting to look into how to send app logs as events to Honeycomb.

**Reply by gerhard** [reply]
WDYT? https://github.com/thechangelog/changelog.com/pull/456

**Reply by gerhard** [reply]
I know that we have replaced Algolia with Typesense for search.

Should we remove all Algolia-related code & secrets @jerodsanto? FWIW: https://github.com/search?q=repo%3Athechangelog%2Fchangelog.com%20algolia&type=code

Since we are talking about search, one thing that I would really like us to improve is the ordering of the search results. I would find most recent entries at the top most useful. For example, for https://changelog.com/search?q=llm, I would have expected for [**LLMs break the internet**](https://changelog.com/podcast/534) to have been the first search result, but instead this is what I'm seeing:

<a href="https://changelog.com/search?q=llm"><img width="1440" alt="image" src="https://user-images.githubusercontent.com/3342/230787291-dc8df8f8-5ed7-4f39-869a-c775d3f5e700.png"></a>

There are a few more search-related improvement suggestions that I have. Limiting myself to one to see how this goes. cc @jasonbosco

**Reply by jerodsanto** [reply]
We can certainly delete the Algolia code. Search improvements would be great too.

**Comment by jasonbosco** [comment]
@gerhard @jerodsanto Starting a new thread from [here](https://github.com/thechangelog/changelog.com/discussions/452#discussioncomment-5565325) to talk about search relevance: 

> https://changelog.com/search?q=llm

We currently consider a match on the `headline` attribute of a news item as more relevant than a match on the `story` or `fragment` attributes:

https://github.com/thechangelog/changelog.com/blob/60fd7eb97853bec14ad140e4f7621ddd3006edda/lib/changelog_web/controllers/search_controller.ex#L10

This is why a post with the title `An LLM playground you can run on your laptop` gets prioritized over the 2nd result which has `LLM` only in the `fragment` / `story` attribute. 

If two news items have `LLM` in the headline and tie, then we use the `published_at` attribute to break the tie: 

https://github.com/thechangelog/changelog.com/blob/60fd7eb97853bec14ad140e4f7621ddd3006edda/lib/changelog_web/controllers/search_controller.ex#L16

So in your example, to get the 2nd result to rank higher based on recency, we would need to set all attributes to have the same weight...

**Comment by jerodsanto** [comment]
@jasonbosco sorry just getting back around to this. It appears the issue is that `LLM` doesn't match in the headline for `LLMs break the internet`, I'm assuming because of the trailing `s`?

If you run this search, it comes up first as it should (because it's both newer and has a `headline` match than the `an LLM playground` post).

https://changelog.com/search?q=llms

Do non-exact word matches like the above not factor in at all?

**Reply by jasonbosco** [reply]
> It appears the issue is that `LLM` doesn't match in the headline for `LLMs break the internet`, I'm assuming because of the trailing `s`?

> Do non-exact word matches like the above not factor in at all?

Non-exact (prefix) matches are indeed accounted for. For eg, if you search for `LLM` and scroll down by about 11 results, you'll see matches for `LLMs` show up, including the `LLMs break the internet` result. All of the results above that one have an exact match for `LLM`, which is why they are ranked first. 

So the key thing is that exact matches are considered better matches and are ranked first, above non-exact partial matches.
